### ISSUE TITLE
two factor authentication not sending 2FA code

### ISSUE THREAD

**Issue by techgaun** [drdoctr/doctr#197]
I have been trying to get 2FA auth code but it just does not work. I had to get 2FA code from the github.com website.

Something like below seems to send me 2FA code:

```shell
curl -u techgaun -d '{"scopes": ["user:email"], "note": "give me 2FA"}' https://api.github.com/authorizations
```

but the following does not:
```shell
curl -u techgaun https://api.github.com/
```

So seems like its github's API that really needs fixed but I'm going to open an issue here as this has been how the github API behaves. Funny thing is that when I go and check logs after doing `doctr configure`, I can see the two_factor_requested in the log but I never get the 2FA code (I'm using SMS). I have to make the explicit call like the first one to get 2FA code

**Comment by asmeurer** [comment]
So is the problem that we're not requesting a scope in the headers?

I guess this is untested. I use the app generated codes. But perhaps even in that case you would want an SMS fallback (is it worth implemeting that?).

**Comment by techgaun** [comment]
@asmeurer the way I did it is make an additional api call to generate github token.. given that we are requiring 2FA, the call is going to fail but thats the only way the 2FA is working for me.. thats how I implemented on #203 and have been using that.
